modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc

```diff
@@ -87,80 +87,295 @@ $ docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig registr
 
 This allows your `kubeconfig` file to be consumed from inside the running container.
 
-[id="cnf-performing-end-to-end-tests-running-the-latency_tests"]
-=== Running the latency tests
-In {product-title} {product-version}, you can also run latency tests from the CNF-test container. The latency test allows you to set a latency limit so that you can determine performance, throughput, and latency.
+[WARNING]
+====
+You must run the latency tests in Discovery mode. The latency tests can change the configuration of your cluster if you do not run in Discovery mode.
+====
 
-The latency test runs the `oslat` tool, which is an open source program to detect OS level latency. For more information, see the Red Hat Knowledgebase solution link:https://access.redhat.com/solutions/5315541[How to measure OS and hardware latency on isolated CPUs?].
+In {product-title} {product-version}, you can also run latency tests from the CNF-test container. The latency test allows you to validate the node tuning is sufficient for your workload.
 
-By default, the latency tests are disabled. To enable the latency test, you must add the `LATENCY_TEST_RUN` variable and set its value to `true`. For example, `LATENCY_TEST_RUN=true`.
+Three tools measure the latency of the system:
 
-Additionally, you can set the following environment variables for latency tests:
+* `hwlatdetect`
+* `cyclictest`
+* `oslat`
 
-* `LATENCY_TEST_RUNTIME` - Specifies the amount of time (in seconds) that the latency test must run.
-* `OSLAT_MAXIMUM_LATENCY` - Specifies the maximum latency (in microseconds) that is expected from all buckets during the `oslat` test run.
+Each tool has a specific use. Use the tools in sequence to achieve reliable test results.
 
-To perform the latency tests, run the following command:
+. The `hwlatdetect` tool measures the baseline that the bare hardware can achieve. Before proceeding with the next latency test, ensure that the number measured by `hwlatdetect` meets the required threshold because hardware latency spikes cannot be fixed by operating system tuning.
+
+.  The `cyclictest` tool verifies the timer latency after `hwlatdetect` passes validation. The `cyclictest` tool schedules a repeated timer and measures the difference between the desired and the actual trigger times. The difference can uncover basic issues with the tuning caused by interrupts or process priorities.
+
+. The `oslat` tool behaves similarly to a CPU-intensive DPDK application and measures all the interruptions and disruptions to the busy loop that simulates CPU heavy data processing.
+
+By default, the latency tests are disabled. To enable the latency test, you must add the `LATENCY_TEST_RUN` environment variable to the test invocation and set its value to `true`. For example, `LATENCY_TEST_RUN=true`.
+
+The test introduces the following environment variables:
```

Comment:
I would verify that we use this structure, The best I can find is that we have examples starting with "Specify...".

I would revise this list to be "`The LATENCY_TEST_CPUS` flag (or whatever) specifies the number of CPUs"